back/routes/adventure.js

```diff
@@ -49,29 +49,42 @@ async function getMean(experienceId, feedbackId) {
 // ----- Get filtered restaurants ------
 router.post('/restaurants', async (req, res) => {
   const userPreference = await preferenceQuery(req.body.username);
+  // TODO: Implement better algorithm for distance
   let distance = 1000;
   if (userPreference.activePreferences.includes("FTmQoqHFVb")) {
     let index = userPreference.activePreferences.indexOf("FTmQoqHFVb")
     if (userPreference.hasMinimumValue[index]) distance = userPreference.minValues[index] * 1000
   }
+
   const allRestaurants = await getGoogleRestaurants(
     distance,
     req.body.lat,
     req.body.lng,
   );
+  const priority = userPreference.prioritize
   const restaurants = [];
   for (const restaurant of allRestaurants) {
     let passesFilter = true;
     let indexPreference = 0;
+    let score = 0;
     for (const preferenceId of userPreference.activePreferences) {
-      const val = await getMean(restaurant.place_id, preferenceId);
-      if (userPreference.hasMinimumValue[indexPreference] && userPreference.minValues[indexPreference] > val) {
+      const meanReviewScore = await getMean(restaurant.place_id, preferenceId);
+      if (userPreference.hasMinimumValue[indexPreference] && userPreference.minValues[indexPreference] > meanReviewScore) {
         passesFilter = false;
         break;
```

Comment:
Why do we break here instead of `continue`? Did we want to skip just this preference, or all remaining preferences?

Reply:
I wanted to skip all the remaining preferences because if it fails on one parameter, it is not matched